### DISCUSSION TITLE
[Feature request] Rearrange "group by" statements in Elasticsearch query

### DISCUSSION THREAD

**Discussion by walterdolce** [grafana/grafana#56256] [Observability]
At the moment, when creating table views showing data stored on Elasticsearch, it's not possible to rearrange the group by/then by rows in the query section. If you want to rearrange the table columns, one would have to pay attention at the order in which those columns get created. 

If a mistake is made - say you want to swap the position of column n1 with column n5 - we would have to delete all clauses and re-create them back again. It would be really great if the group by/then by clause rows could be arrangeable and could be moved up or down (consequentially changing the position of the columns in the table.

### What Grafana version are you using?
5.2.3

### What datasource are you using?
Elasticsearch

### What OS are you running grafana on?
Off the official Grafana docker image

### What did you do?
Tried to rearrange the "group by" rows inside the query view of a table.

### What was the expected result?
To be able to move "group by" rows up or down when multiple "group by"/"then by" rows are present.

### What happened instead?
I could not do it, causing me to constantly delete/re-create the group by/then by clause.

 
<img width="1804" alt="screen shot 2018-09-19 at 14 17 22" src="https://user-images.githubusercontent.com/6195629/45755828-4ef8d480-bc17-11e8-812b-91e41b64acc1.png">

**Comment by dguendisch** [comment]
Missing this as well. I currently help myself by exporting the dashboard, moving the columns/group-by statements in a text editor and then reimport the dashboard.

**Comment by Kijoki1** [comment]
Pain for us as well, would really appreciate if this can be looked at. Still happening on 7.2.0

**Comment by fictitiious** [comment]
If sb is looking for a solution to only change the order of columns in table panel: use **Transform** -> **Organize fields**

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This looks like something that is achievable via a transformation. Feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find a solution and would like to request the feature.
